interface/forms/procedure_order/common.php

```diff
@@ -28,6 +28,7 @@
 use OpenEMR\Core\Header;
 use OpenEMR\Events\Services\QuestLabTransmitEvent;
 use Symfony\Component\EventDispatcher\EventDispatcher;
+use OpenEMR\Modules\Dorn\DornGenHl7Order;//wanting to keep everything in the module for easy updating. I hope this works, from what I've read it should.
```

Comment:
will leave this script review to the module experts. Hopefully a way to use listeners here. Since don't want core code to relay on Modules being around (if we decide to remove modules from the core codebase, for example).

Reply:
So I'd rather this was an Interface or used some kind of registry in the core code.  As we don't want to have module code (where the module might be disabled) in the core code. One mechanism would be to fire off an event to get the dorn connector.  The connector would need to comply with an interface that is in the core code to make sure it has the correct functions etc.

I think a better (more future proof approach) would be to have in core code a LabProcessorRegistry in the codebase, we fire off an event that allows module writers to grab the registry and then register their own processors.  We then loop through the list of processors and ask canProcessLab($labRequestData).  If the processor returns true then we call the processLab($labRequestData) function.  Using a registry and strategy pattern here would make this work with any kind of future lab processing.  If this doesn't make sense then I can look at stubbing out a PR that has this functionality.

Reply:
@adunsulag I did a quick fix here but you're correct and suggested several good options.
However, the procedure orders interface has been custom hard coded considerably over the years with me the last to integrated custom workflow extensively so for now I'm too lazy to implement even more module features just yet. 

Let me know if you find my solution unacceptable or have better idea.
I'm waiting to push until I have a few other fixes done.